src/managers/GuildChannelManager.js

```diff
@@ -126,6 +127,36 @@ class GuildChannelManager extends BaseManager {
     });
     return this.client.actions.ChannelCreate.handle(data).channel;
   }
+
+  /**
+   * Obtains one or more guild channels from Discord, or the channel cache if they're already available.
+   * @param {Snowflake} [id] ID of the channel
+   * @param {boolean} [cache=true] Whether to cache the new channel objects if it weren't already
+   * @param {boolean} [force=false] Whether to skip the cache check and request the API
+   * @returns {Promise<?GuildChannel|Collection<Snowflake, GuildChannel>>}
+   * @example
+   * // Fetch all channels from the guild
+   * message.guild.channels.fetch()
+   *   .then(channels => console.log(`There are ${channels.size} channels.`))
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single channel
+   * message.guild.channels.fetch('222197033908436994')
+   *   .then(channel => console.log(`The channel name is: ${channel.name}`))
+   *   .catch(console.error);
+   */
+  async fetch(id, cache = true, force = false) {
+    if (id && !force) {
+      const existing = this.cache.get(id);
+      if (existing) return existing;
+    }
+
+    // We cannot fetch a single guild channel, as of this commit's date, Discord API throws with 404
```

Comment:
What about `GET /channels/:id`?
It seems to return the same data as `GET /guilds/:id/channels` (except it's only the requested channel instead of all of them).

Reply:
Yeah, that seems to be possible. But however it's quite inconsistent though? the method suggests `GET /guilds/channels/:id` which apparently does not support getting a single ID. Same with `RoleManager#fetch`. But I'll go with it if it looks fine.